src/sinks/elasticsearch/common.rs

```diff
@@ -197,3 +250,38 @@ pub async fn sign_request(
 ) -> crate::Result<()> {
     crate::aws::sign_request("es", request, credentials_provider, region).await
 }
+
+async fn get(
+    base_url: &str,
+    http_auth: &Option<Auth>,
+    aws_auth: &Option<SharedCredentialsProvider>,
+    region: &Option<Region>,
+    request: &RequestConfig,
+    client: HttpClient,
+    path: &str,
+) -> crate::Result<Response<Body>> {
+    let mut builder = Request::get(format!("{}{}", base_url, path));
+
+    if let Some(authorization) = &http_auth {
+        builder = authorization.apply_builder(builder);
+    }
+
+    for (header, value) in &request.headers {
+        builder = builder.header(&header[..], &value[..]);
+    }
+
+    let mut request = builder.body(Bytes::new())?;
+
+    if let Some(credentials_provider) = aws_auth {
+        sign_request(&mut request, credentials_provider, region).await?;
+    }
+    client
+        .send(request.map(hyper::Body::from))
+        .await
+        .map_err(Into::into)
+}
+
+#[derive(Deserialize)]
+struct ClusterState {
+    version: usize,
+}
```

Comment:
Since this struct is only used within the determination of the version when `auto` is specified, it could be constrained to the scope of the helper function referenced in earlier comment.